apps/remote_control/lib/lexical/remote_control/search/indexer/source/reducer.ex

```diff
@@ -108,9 +113,16 @@ defmodule Lexical.RemoteControl.Search.Indexer.Source.Reducer do
   end
 
   defp push_block(%__MODULE__{} = reducer, %Block{} = block) do
-    parent = List.first(reducer.blocks)
+    parent = current_block(reducer)
     block = %Block{block | parent_id: parent.id}
-    %__MODULE__{reducer | blocks: [block | reducer.blocks]}
+    id_path = Enum.reduce(reducer.blocks, [], fn block, acc -> [block.id | acc] end)
+
+    hierarchy =
```

Comment:
Considering that the struct of the block is very simple, what I would like to know is what can I do with `blocks` and `block_hierarchy` separately?

Reply:
For example, to extract references of variables, if the definition is in the parent block, do I need to use the `store.backend` API to find the definition when extracting them?

Reply:
That's correct. 
think of it this way, to search for a variable's instances, you need to start at the reference you're selecting, then find all references to it in each parent block until you get to its definition's block. Then, you need to replace all references in the definition's block, and all of its children.